prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTaskTest.java

```diff
@@ -0,0 +1,120 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class KubeapiWorkFlowTaskTest {
+
+	private final static ApiClient apiClient = Mockito.mock(ApiClient.class);
+
+	private final static KubernetesApi api = Mockito.mock(KubernetesApi.class);
+
+	private final static KubeapiWorkFlowTask task = new KubeapiWorkFlowTask(apiClient, api, "Test");
+
+	@Test
+	public void missingArgs() {
+		WorkContext ctx = createWorkContext(new HashMap<>());
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(MissingParameterException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void invalidOperationType() {
+		HashMap<String, String> map = new HashMap<>();
+		map.put("operation", "test");
+		WorkContext ctx = createWorkContext(map);
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.FAILED, result.getStatus());
+		assertEquals(IllegalArgumentException.class, result.getError().getClass());
+	}
+
+	@Test
+	public void get() throws ApiException {
+		String objJson = "{\"kind\":\"ConfigMap\",\"apiVersion\":\"v1\",\"metadata\":{\"name\":\"my-cm\",\"namespace\":"
+				+ "\"test\",\"uid\":\"5f28eebc-05b2-41b0-9da1-e7399adc1262\",\"resourceVersion\":\"163811\","
+				+ "\"creationTimestamp\":\"2023-04-16T18:47:25Z\"},\"data\":{\"a\":\"vala\",\"b\":\"valb\"}}";
+		DynamicKubernetesObject obj = Dynamics.newFromJson(objJson);
+		WorkContext ctx = workContextGET();
+		doReturn(obj).when(api).get(any(), any(), any(), any(), any(), any());
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.COMPLETED, result.getStatus());
+		assertEquals(objJson, ctx.get("testkey"));
+	}
+
+	@Test
+	public void create() throws ApiException {
+		DynamicKubernetesObject expectedObj = Dynamics.newFromJson(
+				"{\"apiVersion\": \"v1\",\"data\": {\"a\": \"vala\",\"b\": \"valb\"},\"kind\": \"ConfigMap\",\"metadata\": {\"name\": \"my-cm-create\",\"namespace\": \"test\"}}");
+		WorkContext ctx = workContextCREATE();
+		WorkReport result = task.execute(ctx);
+		assertEquals(WorkStatus.COMPLETED, result.getStatus());
+		verify(api, times(1)).create(any(), any(), any(), any(), eq(expectedObj));
+	}
+
+	@Test
+	public void update() throws ApiException {
+		DynamicKubernetesObject expectedObj = Dynamics.newFromJson(
+				"{\"apiVersion\": \"v1\",\"data\": {\"a\": \"valb\",\"b\": \"vala\"},\"kind\": \"ConfigMap\",\"metadata\": {\"resourceVersion\": \"1000\", \"name\": \"my-cm-create\",\"namespace\": \"test\"}}");
+		DynamicKubernetesObject getReturnValue = Dynamics.newFromJson(
+				"{\"apiVersion\": \"v1\",\"data\": {\"a\": \"vala\",\"b\": \"valb\"},\"kind\": \"ConfigMap\",\"metadata\": {\"resourceVersion\": \"1000\", \"name\": \"my-cm-create\",\"namespace\": \"test\"}}");
+		WorkContext ctx = workContextUPDATE();
+		doReturn(getReturnValue).when(api).get(any(), any(), any(), any(), any(), any());
```

Comment:
same here.

Reply:
we do it in other tests as well. Actually, Eloy suggested I use it in my previous assignment. It makes the test code much easier to manage.